examples/phi3/phi3_vision.py

```diff
@@ -0,0 +1,315 @@
+
+    output_dir = Path(args.output_dir).resolve()
+    output_dir.mkdir(parents=True, exist_ok=True)
+    cache_dir = Path(args.cache_dir).resolve()
+    cache_dir.mkdir(parents=True, exist_ok=True)
+
+    if args.inference and args.optimized_model_path:
+        generate(args.optimized_model_path)
+        return
+
+    input_model_path = output_dir / "phi3-vision-128k-instruct" / "pytorch"
+    if not is_model_ready(input_model_path):
+        print(f"Model not found from {input_model_path}, preparing the model...")
+        # prepare the input model
+        run_subprocess(
+            [
+                "bash",
+                "vision/scripts/prepare_phi3_vision_for_olive.sh",
+                str(output_dir),
+            ],
+            env=os.environ,
+        )
+    # if device is gpu, rewrite the genai_config.json
+    genai_config_path = output_dir / "genai_config.json"
+    with genai_config_path.open("r") as f:
+        genai_config = json.load(f)
+    if args.target == "cuda":
+        genai_config["model"]["decoder"]["session_options"]["provider_options"] = [{"cuda": {}}]
+    else:
+        genai_config["model"]["decoder"]["session_options"]["provider_options"] = []
+    with genai_config_path.open("w") as f:
+        json.dump(genai_config, f, indent=4)
+
+    text_embedding_config = generate_text_embedding_config(args, input_model_path)
+    run_and_save(text_embedding_config, output_dir, "text-embedding")
+
+    # Generate Olive configuration file for specific target
+    print("\nGenerating Olive configuration file...")
+    vision_config = generate_vision_config(args, input_model_path)
+    run_and_save(vision_config, output_dir, "vision")
+
+    text_config = generate_text_config(args, input_model_path)
+    try:
+        run_and_save(text_config, output_dir, "text")
+    except Exception:
+        config_link = "https://huggingface.co/microsoft/Phi-3-vision-128k-instruct-onnx-cpu/tree/main/cpu-int4-rtn-block-32-acc-level-4."
+        print(
+            "Ignore the error during Olive run.\nThis script will copy the "
+            "genai_config.json and processor_config.json from ",
+            config_link,
+        )
+        # even if the Olive run fails, partially generated model
+        # are still useful. Copy the model file to the output directory
+        text_config = RunConfig.parse_obj(text_config)
+        cache_dir = text_config.engine.cache_dir
+        pass_config = text_config.passes["builder"].config
+        # find the model file in the cache directory
+        model_path = list(cache_dir.rglob("*.onnx"))[-1]
+        resave_onnx_model(model_path, output_dir / "phi-3-v-128k-instruct-text.onnx", pass_config)
+    print("Model generation completed, output saved to ", output_dir)
+    # clean up the output directory in olive.workflows
+    to_remove_folders = [
+        Path(args.output_dir).resolve() / "vision",
+        Path(args.output_dir).resolve() / "text",
```

Comment:
text_embedding is missing here.

Otherwise, everything else looks good to me!

Reply:
Updated.